bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -0,0 +1,247 @@
+        if (fastPollingScheduler == null || fastPollingScheduler.isCancelled()) {
+            fastPollingScheduler = scheduler.scheduleWithFixedDelay(this::fastPollingSchedulerExecute,
+                    FAST_POLL_INTERVAL, FAST_POLL_INTERVAL, TimeUnit.SECONDS);
+        }
+
+        // start a fast polling burst to ensure the NeHub is initialized quickly
+        startFastPollingBurst();
+    }
+
+    @Override
+    public void dispose() {
+        if (logger.isDebugEnabled())
+            logger.debug("stop background polling..");
+
+        // clean up the lazy polling scheduler
+        if (lazyPollingScheduler != null && !lazyPollingScheduler.isCancelled()) {
+            lazyPollingScheduler.cancel(true);
+            lazyPollingScheduler = null;
+        }
+
+        // clean up the fast polling scheduler
+        if (fastPollingScheduler != null && !fastPollingScheduler.isCancelled()) {
+            fastPollingScheduler.cancel(true);
+            fastPollingScheduler = null;
+        }
+    }
+
+    /*
+     * device handlers call this to initiate a burst of fast polling requests (
+     * improves response time to users when OpenHAB changes a channel value )
+     */
+    public void startFastPollingBurst() {
+        fastPollingCallsToGo.set(FAST_POLL_CYCLES);
+    }
+
+    /*
+     * device handlers call this method to issue commands to the NeoHub
+     */
+    public synchronized NeoHubReturnResult toNeoHubSendChannelValue(String commandStr) {
+        if (socket == null || config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "hub not initialized, status => offline!");
+            return NeoHubReturnResult.ERR_INITIALIZATION;
+        }
+
+        if (logger.isDebugEnabled())
+            logger.debug("sending command {}", commandStr);
+
+        if (socket.sendMessage(commandStr) == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "communication error, status => offline!");
+            return NeoHubReturnResult.ERR_COMMUNICATION;
+        }
+
+        // start a fast polling burst (to confirm the status change)
+        startFastPollingBurst();
+
+        return NeoHubReturnResult.SUCCEEDED;
+    }
+
+    /*
+     * sends a JSON "INFO" request to the NeoHub returns a JSON string that contains
+     * the full status of all devices
+     */
+    protected @Nullable NeoHubInfoResponse fromNeoHubFetchPollingResponse() {
+        if (socket == null || config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "hub not initialized, status => offline!");
+            return null;
+        }
+
+        @Nullable
+        String response = socket.sendMessage(CMD_CODE_INFO);
+        if ((response == null) || response.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
```

Comment:
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Reply:
I added details from the try catch block.